packages/grid/src/vaadin-grid.js

```diff
@@ -655,6 +665,14 @@ class Grid extends ElementMixin(
     cell.id = slotName.replace('-content-', '-');
     cell.setAttribute('tabindex', '-1');
     cell.setAttribute('role', tagName === 'td' ? 'gridcell' : 'columnheader');
+    cell.addEventListener('mouseenter', (e) => {
```

Comment:
Would it make sense from a performance-perspective to only add those listener and dispatch events if there is a custom cell-tooltip generator registered?

Reply:
Getting all the cell references after they've been created may be challenging, so dynamically adding/removing the listeners could be tricky. But dispatching the events at least could be conditional.

Reply:
Updated the PR to only fire events when `vaadin-tooltip` element is added to `vaadin-grid` in the DOM.